content/sensu-go/6.1/plugins/plugins.md

```diff
@@ -0,0 +1,92 @@
+type: "reference"
+description: "Sensu plugins provide executable scripts or other programs that can be used as a Sensu check command, pipe handler command, or mutator command. Read the plugin reference to learn about the Sensu plugin specification."
+weight: 80
+version: "6.1"
+product: "Sensu Go"
+menu:
+  sensu-go-6.1:
+    parent: plugins
+---
+
+Sensu plugins provide executable scripts or other programs that you can use as Sensu checks, handlers, and mutators.
+Sensu plugins must comply with the following specification:
+
+- Accept input/data via `STDIN` (handler and mutator plugins only)
+  - Optionally able to parse a JSON data payload (i.e. observation data in an event)
+- Output data to `STDOUT` or `STDERR`
+- Produce an exit status code to indicate state:
+  - `0` indicates `OK`
+  - `1` indicates `WARNING`
+  - `2` indicates `CRITICAL`
+  - exit status codes other than `0`, `1`, or `2` indicate an unknown or custom
+    status
+- Optionally able to parse command line arguments to modify plugin behavior
+
+## Supported programming languages
+
+You can use any programming language that can satisfy the Sensu plugin specification requirements &mdash; which is nearly any programming language in the world &mdash; to write Sensu plugins.
+
+Using plugins written in programming languages other than Go requires you to install the corresponding runtime.
+For example, to use a Ruby plugin with Sensu Go, you must install the [Sensu Go Ruby Runtime asset][3].
+
+## Plugin execution
+
+All plugins are executed by the Sensu backend.
+Plugins must be executable files that are discoverable on the Sensu system (i.e. installed in a system `$PATH` directory) or referenced with an absolute path (e.g. `/opt/path/to/my/plugin`).
+
+{{% notice note %}}
+**NOTE**: By default, Sensu installer packages will modify the system `$PATH` for the Sensu processes to include `/etc/sensu/plugins`.
+As a result, executable scripts (e.g. plugins) located in `/etc/sensu/plugins` will be valid commands.
+This allows command attributes to use relative paths for Sensu plugin commands, such as `"command": "check-http.rb -u https://sensuapp.org"`.
+{{% /notice %}}
+
+## Tools, templates, and examples
+
+### Go plugin tool and templates
+
+Use the [Sensu Plugin Tool][4] along with a [default template][5] to create the scaffolding for check, handler, mutator, and sensuctl plugins in the Go programming language.
+Follow the instructions in the Sensu Plugin Tool repository to generate a new plugin project based on the template repositories.
+The plugin template repositories wrap the [Sensu Plugin SDK][8], which provides the framework for building Sensu Go plugins.
+
+For a step-by-step walkthrough, read [How to publish an asset with the Sensu Go SDK][7] &mdash; you'll learn how to create a check plugin and a handler plugin with the Sensu Plugin SDK.
+You can also watch our 30-minute webinar, [Intro to assets with the Sensu Go SDK][6], and learn to build a check plugin.
+
```

Comment:
Would it be a good idea to add the example I created using the Sensu Plugin Tool here? --->

{{< code go >}}
package main

import (
	"fmt"
	"log"

	"github.com/sensu-community/sensu-plugin-sdk/sensu"
	"github.com/sensu/sensu-go/types"
)

// Config represents the check plugin config.
type Config struct {
	sensu.PluginConfig
	Example string
}

var (
	plugin = Config{
		PluginConfig: sensu.PluginConfig{
			Name:     "check_name",
			Short:    "Description for check_name",
			Keyspace: "sensu.io/plugins/check_name/config",
		},
	}

	options = []*sensu.PluginConfigOption{
		&sensu.PluginConfigOption{
			Path:      "example",
			Env:       "CHECK_EXAMPLE",
			Argument:  "example",
			Shorthand: "e",
			Default:   "",
			Usage:     "An example string configuration option",
			Value:     &plugin.Example,
		},
	}
)

func main() {
	check := sensu.NewGoCheck(&plugin.PluginConfig, options, checkArgs, executeCheck, false)
	check.Execute()
}

func checkArgs(event *types.Event) (int, error) {
	if len(plugin.Example) == 0 {
		return sensu.CheckStateWarning, fmt.Errorf("--example or CHECK_EXAMPLE environment variable is required")
	}
	return sensu.CheckStateOK, nil
}

func executeCheck(event *types.Event) (int, error) {
	log.Println("executing check with --example", plugin.Example)
	return sensu.CheckStateOK, nil
}
{{< /code >}}

...Or is this not the right file to use as an example?

Reply:
I am just going to go for it and try to add part of this as an example on the theory that it's usually easier to point out what's wrong.